src/cli/command_info.rs

```diff
@@ -4,7 +4,100 @@ use miette::{Context, IntoDiagnostic};
 use std::env;
 use std::path::PathBuf;
 
-use super::{Command, get_styles};
+use super::{Args, Command, get_styles};
+
+/// Get all built-in command names including aliases (discovered dynamically from clap)
+fn get_builtin_commands_with_aliases() -> Vec<String> {
+    let mut commands = Vec::new();
+
+    for subcommand in Args::command().get_subcommands() {
+        // Add main command name
+        commands.push(subcommand.get_name().to_string());
+
+        // Add all aliases
+        commands.extend(subcommand.get_all_aliases().map(|alias| alias.to_string()));
+    }
+
+    commands
+}
+
+/// Get all external command names (discovered from PATH)
+fn get_external_command_names() -> Vec<String> {
+    find_external_commands()
+        .into_iter()
+        .map(|(name, _path)| name)
+        .collect()
+}
+
+/// All available commands (built-in + external)
+fn get_all_available_commands() -> Vec<String> {
+    let mut all_commands = Vec::new();
+
+    all_commands.extend(get_builtin_commands_with_aliases());
+
+    all_commands.extend(get_external_command_names());
+
+    all_commands.sort();
+    all_commands.dedup();
+
+    all_commands
+}
+
+/// Find similar commands using Jaro similarity
+fn find_similar_commands(input: &str, threshold: f64) -> Vec<String> {
+    let available_commands = get_all_available_commands();
+    let mut suggestions: Vec<(f64, String)> = Vec::new();
+
+    for command in available_commands {
+        let similarity = strsim::jaro(input, &command);
+        if similarity > threshold {
+            suggestions.push((similarity, command));
+        }
+    }
+
+    // Sort by similarity (ascending), take top 3, reverse to get most similar first
+    suggestions.sort_by(|a, b| a.0.partial_cmp(&b.0).unwrap_or(std::cmp::Ordering::Equal));
```

Comment:
let's sort them descending, so the most similar is the bottom 3. and you return entire list.

in the CLI itself, you will just do `if let Some(most_similar) = suggestions.pop`

this way you can remove len == 1 check and maintain more easier API to use